### PR TITLE
fix: support karpenter version > 0.19.x

### PR DESCRIPTION
**Pull request by solidnerd**
### Description

During the release of karpenter 0.19.x there were breaking change in the configuration

https://github.com/aws/karpenter/commit/0b122ce5d3b03289300be596163a895994c01713

To support all version mappings it check for the version of karpenter release to install

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Review by mbevc1 — APPROVED** [review]

File: pkg/karpenter/karpenter.go [hunk 1]
```diff
@@ -67,6 +70,11 @@ func (k *Installer) Install(ctx context.Context, serviceAccountRoleARN string, i
 		},
 		serviceAccountName: DefaultServiceAccountName,
 	}
+
+	versions := strings.Split(k.ClusterConfig.Karpenter.Version, ".")
+
+	minor, err := strconv.Atoi(versions[1])
+
```
Comment (Himangini) [inline pkg/karpenter/karpenter.go:77]: Please use a version library to compare versions.

File: pkg/karpenter/karpenter.go [hunk 2]
```diff
@@ -76,6 +84,19 @@ func (k *Installer) Install(ctx context.Context, serviceAccountRoleARN string, i
 		serviceAccount: serviceAccountMap,
 	}
 
+	if minor >= 19 {
+		values = map[string]interface{}{
```
Comment (Himangini) [inline pkg/karpenter/karpenter.go:88]: shouldn't lines 78 to 85 be part of this if's else?

**Review by Himangini — CHANGES_REQUESTED** [review]
Please add tests to cover your changes, also see comments below.

**Comment by mbevc1** [comment]
We should probably bump it to 0.21.x

**Pull request closed without merging**